astronomer/providers/amazon/aws/triggers/emr.py

```diff
@@ -71,6 +71,113 @@ async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+    :param name: The name of the job run.
+    :param execution_role_arn: The IAM role ARN associated with the job run.
+    :param release_label: The Amazon EMR release version to use for the job run.
+    :param job_driver: Job configuration details, e.g. the Spark job parameters.
+    :param configuration_overrides: The configuration overrides for the job run,
+            specifically either application configuration or monitoring configuration.
+    :param client_request_token: The client idempotency token of the job run request.
+    :param aws_conn_id: Reference to AWS connection id.
+    :param poll_interval: polling period in seconds to check for the status.
+    :param max_retries: maximum retry for poll for the status.
+    """
+
+    INTERMEDIATE_STATES: List[str] = ["PENDING", "SUBMITTED", "RUNNING"]
+    FAILURE_STATES: List[str] = ["FAILED", "CANCELLED", "CANCEL_PENDING"]
+    SUCCESS_STATES: List[str] = ["COMPLETED"]
+
+    def __init__(
+        self,
+        name: str,
+        virtual_cluster_id: str,
+        execution_role_arn: str,
+        release_label: str,
+        job_driver: Dict[str, Any],
+        configuration_overrides: Optional[Dict[str, Any]] = None,
+        client_request_token: Optional[str] = None,
+        aws_conn_id: str = "aws_default",
+        poll_interval: int = 30,
+        max_tries: Optional[int] = None,
+    ):
+        super().__init__()
+        self.virtual_cluster_id = virtual_cluster_id
+        self.name = name
+        self.execution_role_arn = execution_role_arn
+        self.release_label = release_label
+        self.job_driver = job_driver
+        self.configuration_overrides = configuration_overrides
+        self.client_request_token = client_request_token
+        self.aws_conn_id = aws_conn_id
+        self.poll_interval = poll_interval
+        self.max_tries = max_tries
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        """Serializes EmrContainerOperatorTrigger arguments and classpath."""
+        return (
+            "astronomer.providers.amazon.aws.triggers.emr.EmrContainerOperatorTrigger",
+            {
+                "virtual_cluster_id": self.virtual_cluster_id,
+                "name": self.name,
+                "execution_role_arn": self.execution_role_arn,
+                "release_label": self.release_label,
+                "job_driver": self.job_driver,
+                "configuration_overrides": self.configuration_overrides,
+                "client_request_token": self.client_request_token,
+                "aws_conn_id": self.aws_conn_id,
+                "max_tries": self.max_tries,
+                "poll_interval": self.poll_interval,
+            },
+        )
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        """Run until EMR container reaches the desire state"""
+        hook = EmrContainerHookAsync(aws_conn_id=self.aws_conn_id, virtual_cluster_id=self.virtual_cluster_id)
+        try:
```

Comment:
This is not a heavy call since it does not wait for resources to come up instead it immediately returns the metadata so keeping it in trigger should not be problematic. Also, I don't want to initialise trigger both in hook and operator.